tests/testthat/test_calc_age.R

```diff
@@ -0,0 +1,82 @@
+#### Tests for calc_age function
+
+library(dplyr)
+
+#### Old dummy data from the calc_age function
+# set.seed(666)
+# bird_data <-
+#   tibble::tibble(IndvID = LETTERS[sample(1:26, 100, replace = TRUE)],
+#                  SampleYear = sample(2012:2016, 100, replace = TRUE),
+#                  CaptureDate = as.Date(paste(SampleYear, 3, 31, sep = "-"), format = "%Y-%m-%d")
+#                                + sample(1:30, 100, replace = TRUE),
+#                  Age_obsv = sample(c(1L, 4L), 100, replace = TRUE)) %>%
+#   mutate(Age_calculated = calc_age(data = ., ID = IndvID, Age = Age_obsv,
+#                                    Date = CaptureDate, Year = SampleYear))
+
+
+#### Create dummy dataset of Capture data and run the calc_age function
+set.seed(42)
+Capture_data <-
+  tibble::tibble(
+    #### include some individuals without ID
+    IndvID = sample(c(paste0("A", 1:25), NA), replace = TRUE, size = 100),
+    Species = "PARMAJ") %>%
+  dplyr::mutate(SampleYear = sample(2011:2018, 100, replace = TRUE),
+                CaptureDate = as.Date(paste(SampleYear, 3, 31, sep = "-"), format = "%Y-%m-%d")
+                + sample(1:30, 100, replace = TRUE)) %>%
+  dplyr::arrange(IndvID, CaptureDate) %>%
+  group_by(IndvID) %>%
+  #### assign to some randomly selected IDs code 1 as first Age_obsv, others 4
+  dplyr::mutate(rowid_group = row_number(),
+                Age_obsv = case_when(IndvID %in% c("A12", "A30", "A21", "A23",
+                                                   "A6", "A8", "A10", "A17",
+                                                   "A20", "A2", "A18", "A5") &
+                                       rowid_group == 1 ~ 1L,
+                                     TRUE ~ 4L)) %>%
```

Comment:
This will produce many individuals which exhibit the same behaviour from our function (i.e. either caught as nestling or as adult). For the tests these records are redundant because they're not being explicitly tested. It would be better to create a small number of individuals that fit different possible scenarios and are all tested (see my comment below) rather than many individuals that are not tested.